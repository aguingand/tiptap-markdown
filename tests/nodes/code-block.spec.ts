import { expect, test } from "vitest";
import { parse, serialize } from "../utils";


test('parse markdown', () => {
    expect(parse('```\nexample\n```')).toMatchSnapshot();
});
test('parse markdown with lang', () => {
    expect(parse('```js\nexample\n```')).toMatchSnapshot();
});
test('parse markdown with languageClassPrefix', () => {
    expect(parse('```js\nexample\n```', { codeBlock: { languageClassPrefix: 'lang--' } }, true))
        .toEqual('<pre><code class="lang--js">example\n</code></pre>');
})
test('parse markdown with indent', () => {
    expect(parse('    example')).toMatchSnapshot();
});
test('parse html', () => {
    expect(parse('<pre><code>example</code></pre>')).toMatchSnapshot();
});
test('serialize', () => {
    expect(serialize('<pre><code>example</code></pre>')).toEqual('```\nexample\n```\n');
});
test('serialize with lang', () => {
    expect(serialize('<pre><code class="language-js">example</code></pre>')).toEqual('```js\nexample\n```\n');
});
test('serialize with languageClassPrefix', () => {
    expect(serialize('<pre><code class="lang--js">example</code></pre>', { codeBlock: { languageClassPrefix: 'lang--' } }))
        .toEqual('```js\nexample\n```\n');
});
