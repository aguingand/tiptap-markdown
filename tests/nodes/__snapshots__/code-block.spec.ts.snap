// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse html 1`] = `
[
  {
    "attrs": {
      "language": null,
    },
    "content": [
      {
        "text": "example",
        "type": "text",
      },
    ],
    "type": "codeBlock",
  },
]
`;

exports[`parse markdown 1`] = `
[
  {
    "attrs": {
      "language": null,
    },
    "content": [
      {
        "text": "example
",
        "type": "text",
      },
    ],
    "type": "codeBlock",
  },
]
`;

exports[`parse markdown with indent 1`] = `
[
  {
    "attrs": {
      "language": null,
    },
    "content": [
      {
        "text": "example
",
        "type": "text",
      },
    ],
    "type": "codeBlock",
  },
]
`;

exports[`parse markdown with lang 1`] = `
[
  {
    "attrs": {
      "language": "js",
    },
    "content": [
      {
        "text": "example
",
        "type": "text",
      },
    ],
    "type": "codeBlock",
  },
]
`;

exports[`parse markdown with languageClassPrefix 1`] = `
[
  {
    "attrs": {
      "language": "js",
    },
    "content": [
      {
        "text": "example
",
        "type": "text",
      },
    ],
    "type": "codeBlock",
  },
]
`;
